src/pages/products-page/Main.jsx

```diff
@@ -1,108 +1,104 @@
+import axios from "axios";
+import { useEffect } from "react";
 import img3 from "../../assets/images/Jeans1.JPG";
-export default function Main() {
-  return (
-    <main>
-      <div class="h-main-heading">
-        Showing All Proucts <small>(Showing 20 products)</small>
-      </div>
-      <div class="ha-grid-main">
-        <div class="h-grid-product">
-          <img src={img3} alt="" />
-          <div class="h-product-icon wish-listed">
-            <i class="fas fa-heart"></i>
-          </div>
-
-          <div class="h-product-bottom">
-            <p class="product-name">Jeans</p>
-            <p class="product-desc">Washed Slim Fit Jeans</p>
-            <p class="product-desc">
-              Offer Price: <small>Rs.12,60</small>
-            </p>
-
-            <button class="h-product-button">Add to Cart</button>
-          </div>
-        </div>
-        <div class="h-grid-product">
-          <img src={img3} alt="" />
-          <div class="h-product-icon wish-listed">
-            <i class="fas fa-heart"></i>
-          </div>
+import { products } from "../../backend/db/products";
+import { useProduct } from "../../product-context";
+import ProductCard from "./ProductCard";
 
-          <div class="h-product-bottom">
-            <p class="product-name">Jeans</p>
-            <p class="product-desc">Washed Slim Fit Jeans</p>
-            <p class="product-desc">
-              Offer Price: <small>Rs.12,60</small>
-            </p>
-
-            <button class="h-product-button">Add to Cart</button>
-          </div>
-        </div>
-        <div class="h-grid-product">
-          <img src={img3} alt="" />
-          <div class="h-product-icon">
-            <i class="fas fa-heart"></i>
-          </div>
-
-          <div class="h-product-bottom">
-            <p class="product-name">Jeans</p>
-            <p class="product-desc">Washed Slim Fit Jeans</p>
-            <p class="product-desc">
-              Offer Price: <small>Rs.12,60</small>
-            </p>
+export default function Main() {
+  const { state, dispatch } = useProduct();
 
-            <button class="h-product-button">Add to Cart</button>
-          </div>
-        </div>
+  useEffect(() => {
+    fetch("api/products")
+      .then((response) => response.json())
+      .then((data) =>
+        dispatch({ type: "SET_PRODUCTS", payload: data.products })
+      );
+  }, []);
 
-        <div class="h-grid-product">
-          <img src={img3} alt="" />
-          <div class="h-product-icon">
-            <i class="fas fa-heart"></i>
-          </div>
+  function sortingByPrice(items, sort) {
```

Comment:
Keep the function **pure**, don't modify incoming data. This can be done in a short way and more readable way:
`
[...items].sort((a,b) => sort === "LOW_TO_HIGH") ? Number(a.price) - Number(b.price)): Number(b.price) - Number(a.price));
`

Reply:
yes you are right, will improve this one also.